streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -662,6 +662,9 @@ private void maybeSetRunning() {
                 return;
             }
 
+            // all (alive) threads have received their assignment, close any remaining "pending" Tasks, they're not needed
+            stateDirectory.closePendingTasks();
```

Comment:
I am not even sure you need this if you close the tasks that are not needed during the first assignment. A check whether there still are some "dummy" tasks might still be useful.

Reply:
If there are multiple threads on the same instance, we only want to close the "not needed" pending task once they have _all_ completed their assignment.